### PR TITLE
Close 'pop ups' (sheet) if the user presses escape or clicks outside the box of presses escape

### PR DESCRIPTION
**Pull request by Happy0**
Sometimes when I'm using patchwork and I'm in a view that has a pop up (such as the 'join pub' or 'edit profile' views), I want to quickly get out of that view by clicking outside the box or pressing escape. However, currently the only way to escape the pop up is to click the 'cancel' button.

This pull request makes it possible to dismiss the pop up (called 'sheet' in the code base) by clicking outside of it or pressing escape.

Not sure if you / others expect this behaviour too - I thought I'd make a pull request and give you the option :).

### CODE REVIEW

**Comment by Happy0** [comment]
@mmckegg - just pre-emptively bumping this before stalebot breaths down my neck :P. Take a look at this pull request when you have the time.

**Comment by mmckegg** [comment]
Hey @Happy0,

I'm happy to merge the esc part ASAP, but I think more discussion is needed about whether clicking out should close. **If you remove that part, and just leave the esc handling, I'll merge.** Then maybe create a seperate PR for handling clicking out where we can discuss?

**Comment by christianbundy** [comment]
Just to add my data point: clicking outside to close the dialog is intuitive and expected for me as well.

**Comment by mmckegg** [comment]
@christianbundy only if the sheet is modal though, which it is not. I quite often find myself copying stuff out of other posts into gatherings (without wanting it to close).

This is the aspect that I want to discuss.

**Comment by Happy0** [comment]
@mmckegg - I've updated this PR to remove the 'click outside the box' functionality

File: modules/sheet/display.js [hunk 1]
```diff
@@ -12,12 +12,21 @@ exports.create = function () {
       h('footer', [footer])
     ])
 
+    // Closes the sheet when the user presses escape
+    function escapeKeyListener(event) {
+      event.stopPropagation()
```
Comment (mmckegg) [inline modules/sheet/display.js]: shouldn't this have an `if (event.code == 'Escape')` or something??

**Comment by Happy0** [comment]
@mmckegg - yes, sorry - well spotted. I've fixed it, squashed my commits and updated this PR :)

**Comment by christianbundy** [comment]
@Happy0  Should this remain open?

**Comment by Happy0** [comment]
@christianbundy - i'm not sure when I'll next have time to look at this, so feel free to close it if you like. I can always open a new ticket if i return to it. Sorry about letting it go stale

**Pull request closed without merging**